internal/core/data/v2/controller/http/event_test.go

```diff
@@ -203,3 +228,66 @@ func TestAddEvent(t *testing.T) {
 		})
 	}
 }
+
+func TestEventById(t *testing.T) {
+	expectedMessage := "Get event by id successfully"
+
+	validEventId := expectedEventId
+	emptyEventId := ""
+	invalidEventId := "bad"
+	notFoundEventId := InexistentEventID
+
+	dbClientMock := &dbMock.DBClient{}
+	dbClientMock.On("EventById", notFoundEventId).Return(models.Event{}, errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, "event doesn't exist in the database", nil))
+	dbClientMock.On("EventById", validEventId).Return(persistedEvent, nil)
+
+	dic := mocks.NewMockDIC()
+	dic.Update(di.ServiceConstructorMap{
+		v2DataContainer.DBClientInterfaceName: func(get di.Get) interface{} {
+			return dbClientMock
+		},
+	})
+
+	ec := NewEventController(dic)
+	assert.NotNil(t, ec)
+
+	tests := []struct {
+		Name               string
+		EventId            string
+		ErrorExpected      bool
+		ExpectedStatusCode int
+	}{
+		{"Valid - Find Event by Id", validEventId, false, http.StatusOK},
+		{"Invalid - Empty EventId", emptyEventId, true, http.StatusBadRequest},
+		{"Invalid - EventId is not an UUID", invalidEventId, true, http.StatusBadRequest},
+		{"Invalid - Event doesn't exist", notFoundEventId, true, http.StatusNotFound},
+	}
+
+	for _, testCase := range tests {
+		t.Run(testCase.Name, func(t *testing.T) {
+			reqPath := fmt.Sprintf("%s/%s/%s", contractsV2.ApiEventRoute, contractsV2.Id, testCase.EventId)
+			req, err := http.NewRequest(http.MethodGet, reqPath, http.NoBody)
+			req = mux.SetURLVars(req, map[string]string{contractsV2.Id: testCase.EventId})
+			require.NoError(t, err)
+
+			recorder := httptest.NewRecorder()
+			handler := http.HandlerFunc(ec.EventById)
+			handler.ServeHTTP(recorder, req)
+
+			var actualResponse responseDTO.EventResponse
+			err = json.Unmarshal(recorder.Body.Bytes(), &actualResponse)
+			require.NoError(t, err)
+
+			if testCase.ErrorExpected {
+				assert.Equal(t, contractsV2.ApiVersion, actualResponse.ApiVersion, "API Version not as expected")
```

Comment:
Add require.Error() before asserts

Reply:
not really, the err here is the json parsing error.  In controller, there is no real error returned, and the error message is stated in the response with error http status code.

Reply:
Ok, see it now.